src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php

```diff
@@ -289,10 +289,16 @@ protected function createNotFoundException(string $message = 'Not Found', \Excep
      *
      *     throw $this->createAccessDeniedException('Unable to access this page!');
      *
+     * @throws \LogicException If SecurityBundle is not available
+     *
      * @final since version 3.4
      */
     protected function createAccessDeniedException(string $message = 'Access Denied.', \Exception $previous = null): AccessDeniedException
     {
+        if (!class_exists(AccessDeniedException::class)) {
+            throw new \LogicException('You can not use the "createAccessDeniedException" method if the SecurityBundle is not registered in your application. Try running "composer require security".');
```

Comment:
Only the security-core component is required in fact, should it be more precise?

Reply:
I wondered about that too. On the one hand, previously, "security-core" and "security-csrf" were required, not the complete "security". But on the other hand two current situations where a requirement is suggested suggest the complete "security" (but one suggests "security-csrf"): https://github.com/symfony/framework-bundle/search?utf8=%E2%9C%93&q=composer+require+security&type=

Reply:
we're moving to as-specific-as-possible, so we should update the places where symfony/security is referenced

Reply:
other errors are about missing services so suggesting `security` is correct (alias for security-bundle, not symfony/security). We only need to change for `security-core` on this one